samples/subsys/ipc/openamp/remote/src/main.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2018, NXP
+ * Copyright (c) 2018, Nordic Semiconductor ASA
+ * Copyright (c) 2018, Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ipm.h>
+#include <misc/printk.h>
+#include <device.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <openamp/open_amp.h>
+#include <metal/device.h>
+
+#include "common.h"
+
+#define APP_TASK_STACK_SIZE (1024)
+K_THREAD_STACK_DEFINE(thread_stack, APP_TASK_STACK_SIZE);
+static struct k_thread thread_data;
+
+static struct device *ipm_handle;
+
+static metal_phys_addr_t shm_physmap[] = { SHM_START_ADDR };
+static struct metal_device shm_device = {
+	.name = SHM_DEVICE_NAME,
+	.bus = NULL,
+	.num_regions = 1,
+	{
+		{
+			.virt       = (void *) SHM_START_ADDR,
+			.physmap    = shm_physmap,
+			.size       = SHM_SIZE,
+			.page_shift = 0xffffffff,
+			.page_mask  = 0xffffffff,
+			.mem_flags  = 0,
+			.ops        = { NULL },
+		},
+	},
+	.node = { NULL },
+	.irq_num = 0,
+	.irq_info = NULL
+};
+
+static volatile unsigned int received_data;
+
+static struct virtio_vring_info rvrings[2] = {
+	[0] = {
+		.info.align = VRING_ALIGNMENT,
+	},
+	[1] = {
+		.info.align = VRING_ALIGNMENT,
+	},
+};
+static struct virtio_device vdev;
+static struct rpmsg_virtio_device rvdev;
+static struct metal_io_region *io;
+static struct virtqueue *vq[2];
+
+static unsigned char virtio_get_status(struct virtio_device *vdev)
+{
+	unsigned char *status_byte = (void *)RSC_TABLE_ADDRESS;
+
+	return *status_byte;
+}
+
+static void virtio_set_status(struct virtio_device *vdev, unsigned char status)
+{
+	u32_t dummy_data = 0x00220022; /* Some data must be provided */
```

Comment:
As this example the remote side doesn't set status, it looks like this function can do nothing, don't need to sent notification.

Reply:
In fact this is requested for remote side, called in rpmsg_virtio_wait_remote_ready.:
		if (status & VIRTIO_CONFIG_STATUS_NEEDS_RESET) {
			rpmsg_virtio_set_status(rvdev, 0);
			/* TODO notify remote processor */
		} 
In this case the status would be updated, so suggest to clean the mailbox send and just set status to 0.

Reply:
Since I know we don't ever set VIRTIO_CONFIG_STATUS_NEEDS_RESET I'm just going to remove virtio_set_status from the remote side.